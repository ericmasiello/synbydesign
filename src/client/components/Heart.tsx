import * as React from 'react';
import styled from 'styled-components';
import { COLORS } from '../styles/vars';
import { pxToRem } from '../styles/utils';

export const HeartIcon = styled.svg`
  height: ${pxToRem(30)};
  width: ${pxToRem(30)};
  fill: transparent;
  stroke: ${COLORS.highlight};
  transition-property: fill, transform, filter;
  transition: 0.3s;
`;

HeartIcon.displayName = 'HeartIcon';

interface Props extends React.HTMLProps<HTMLButtonElement> {
  tag?: Tag;
  selected?: boolean;
  className?: string;
}

export const Heart: React.SFC<Props> = props => {
  const { className, tag: Tag, children, ...rest } = props;
  return (
    // @ts-ignore
    <Tag className={className} {...rest}>
      <HeartIcon viewBox="-4.75 11.781 10.5 9.5">
        <path
          strokeWidth="0.75"
          strokeMiterlimit="10"
          d="M-1.7,12.148c-1.587,0-2.664,1.455-2.664,3.059
	c0,2.748,2.136,3.881,4.864,5.645c2.729-1.764,4.864-2.896,4.864-5.645c0-1.604-1.077-3.059-2.664-3.059
	c-1.021,0-1.86,0.572-2.2,0.916C0.16,12.721-0.679,12.148-1.7,12.148z"
        />
      </HeartIcon>
    </Tag>
  );
};

Heart.displayName = 'Heart';

Heart.defaultProps = {
  tag: 'button',
};

export default styled(Heart)`
  display: inline-flex;
  border: none;
  background: none;
  padding: ${pxToRem(10)};

  ${HeartIcon} {
    ${props => (props.selected ? `fill: ${COLORS.highlight}` : '')};
  }

  &:hover,
  &:focus,
  &:active {
    ${HeartIcon} {
      fill: ${COLORS.highlight};
      transform: scale(1.2);
      filter: contrast(1.5);
    }
  }

  &:active {
    ${HeartIcon} {
      transform: scale(1.5);
    }
  }
`;
