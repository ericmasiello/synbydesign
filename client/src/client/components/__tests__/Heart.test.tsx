import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Heart } from '../Heart';

it('should render', () => {
  const component = renderer.create(<Heart />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should display additional classes', () => {
  const wrapper = shallow(<Heart className="my-class" />);

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render additional props', () => {
  const wrapper = shallow(<Heart data-test="my-attribute" />);

  expect(wrapper.find('[data-test="my-attribute"]')).toHaveLength(1);
});

it('should render as a <button> by default', () => {
  const wrapper = shallow(<Heart />);

  expect(wrapper.type()).toBe('button');
});

it('should render as a custom tag', () => {
  const wrapper = shallow(<Heart tag="a" />);

  expect(wrapper.type()).toBe('a');
});

it('should only render a heart icon', () => {
  const wrapper = shallow(<Heart />);

  expect(wrapper.children().name()).toBe('HeartIcon');
});
