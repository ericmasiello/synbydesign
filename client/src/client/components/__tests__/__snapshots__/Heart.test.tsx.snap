// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<button>
  <svg
    className="sc-bdVaJa kxTICB"
    fill="transparent"
    viewBox="-4.75 11.781 10.5 9.5"
  >
    <path
      d="M-1.7,12.148c-1.587,0-2.664,1.455-2.664,3.059
	c0,2.748,2.136,3.881,4.864,5.645c2.729-1.764,4.864-2.896,4.864-5.645c0-1.604-1.077-3.059-2.664-3.059
	c-1.021,0-1.86,0.572-2.2,0.916C0.16,12.721-0.679,12.148-1.7,12.148z"
      strokeMiterlimit="10"
      strokeWidth="0.75"
    />
  </svg>
</button>
`;
